cpp/pybind/io/class_io.cpp

```diff
@@ -194,22 +194,73 @@ void pybind_class_io(py::module &m_io) {
     docstring::FunctionDocInject(m_io, "read_point_cloud",
                                  map_shared_argument_docstrings);
 
+    m_io.def(
+            "read_point_cloud_from_bytes",
+            [](const py::bytes &bytes, const std::string &format,
+               bool remove_nan_points, bool remove_infinite_points,
+               bool print_progress) {
+                const char *dataptr = PYBIND11_BYTES_AS_STRING(bytes.ptr());
+                auto length = PYBIND11_BYTES_SIZE(bytes.ptr());
+                auto buffer = new unsigned char[length];
+                memcpy(buffer, dataptr, length);  // copy before releasing GIL
```

Comment:
same, `std::memcpy`

Reply:
@yxlao We have no choice but to copy before releasing the GIL, we can't release the gil and also operate on a `Py_buffer` as far as I know.

Reply:
Thanks, for a minute I thought you were talking about something else 😆

Reply:
Fixed in e64217f254a3446818d5b51a7a8f549033b80e2f